### PR TITLE
Fix unit label hover title for metric ton

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
This temporarily fixes unit label hover title for metric ton, until we remove metric ton unit from schema and validator unit.json.

### How to review
Run new functional test, check unit tests, where locale is set to "en_GB".

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by petechd** [comment]
[Bug: Metric ton on unit label hover](https://trello.com/c/7CJiBJAv/5764-bug-metric-ton-on-unit-label-hover)

File: app/jinja_filters.py [hunk 1]
```diff
@@ -79,11 +79,17 @@ def format_unit(
     # until removed from schema we substitute mass-tonne for mass-metric-ton before format unit
     measurement_unit = "mass-tonne" if unit == "mass-metric-ton" else unit
 
+    locale = (
+        DEFAULT_LOCALE
+        if str(flask_babel.get_locale()) in {"en", "eo"}
```
Comment (katie-gardner) [inline app/jinja_filters.py:84]: How come this was needed for "eo" as well?

File: app/jinja_filters.py [hunk 2]
```diff
@@ -79,11 +79,17 @@ def format_unit(
...
+        else flask_babel.get_locale()
+    )
+
     formatted_unit: str = units.format_unit(
         value=value,
         measurement_unit=measurement_unit,
         length=length,
-        locale=flask_babel.get_locale(),
+        locale=locale,
```
Comment (MebinAbraham) [inline app/jinja_filters.py:92]: Keep this as it was and update out `get_locale` method instead, then we set it correct in all places.

File: app/jinja_filters.py (hunk 1 above)
Comment (petechd) [reply app/jinja_filters.py:84]: This is a historical code, "eo" was a language code used for Ulsters Scots, as they didn't have an official one, that was supported during census, it uses same transforms as "en" but I guess it is not needed in here, so I will remove it.

**Review by Yuyuutsu — COMMENTED** [review]
When I hover over the unit in `test_unit_label_hover` I am seeing a question mark.

**Comment by petechd** [comment]
> When I hover over the unit in `test_unit_label_hover` I am seeing a question mark.



> When I hover over the unit in `test_unit_label_hover` I am seeing a question mark.

You need to wait a sec longer for the text to appear.

File: app/jinja_filters.py (hunk 2 above)
Comment (petechd) [reply app/jinja_filters.py]: Done, thanks!

File: app/jinja_filters.py [hunk 3]
```diff
@@ -83,7 +84,7 @@ def format_unit(
         value=value,
         measurement_unit=measurement_unit,
         length=length,
-        locale=flask_babel.get_locale(),
+        locale=setup.get_locale(),
```
Comment (petechd) [inline app/jinja_filters.py:87]: Circular import issue hence written this way.

File: schemas/test/en/test_unit_label_hover.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (MebinAbraham) [inline schemas/test/en/test_unit_label_hover.json:1]: Why do you need a schema for this?. Use the existing units one?

File: tests/functional/spec/features/unit_label_hover/unit_label_hover.spec.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+describe("Unit Label Hover", () => {
```
Comment (MebinAbraham) [inline tests/functional/spec/features/unit_label_hover/unit_label_hover.spec.js:1]: Add it to existing unit spec.

**Comment by Yuyuutsu** [comment]
> > When I hover over the unit in `test_unit_label_hover` I am seeing a question mark.
> 
> You need to wait a sec longer for the text to appear.
Oops yeah, I can see it now 👍

File: schemas/test/en/test_unit_label_hover.json (hunk 4 above)
Comment (petechd) [reply schemas/test/en/test_unit_label_hover.json]: Just cause I wanted it separate, removed now.

File: tests/functional/spec/features/unit_label_hover/unit_label_hover.spec.js (hunk 5 above)
Comment (petechd) [reply tests/functional/spec/features/unit_label_hover/unit_label_hover.spec.js]: Added now.

Comment (petechd) [reply tests/functional/spec/features/unit_label_hover/unit_label_hover.spec.js]: As I mentioned, length units were all American en.

**Review by rmccar — APPROVED** [review]

**Review by katie-gardner — APPROVED** [review]
Nicely done 👍

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**